destination.go

```diff
@@ -0,0 +1,210 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	config         DestinationConfig
+	dynamoDBClient *dynamodb.Client
+}
+
+type DestinationConfig struct {
+	sdk.DefaultDestinationMiddleware
+
+	Config
+}
+
+func NewDestination() sdk.Destination {
+	return sdk.DestinationWithMiddleware(&Destination{})
+}
+
+func (d *Destination) Config() sdk.DestinationConfig {
+	return &d.config
+}
+
+func (d *Destination) Open(ctx context.Context) error {
+	sdk.Logger(ctx).Info().Msg("Opening DynamoDB Source...")
+
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion(d.config.AWSRegion),
+		config.WithCredentialsProvider(aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(d.config.AWSAccessKeyID, d.config.AWSSecretAccessKey, ""))),
+	)
+	if err != nil {
+		return fmt.Errorf("could not load AWS config: %w", err)
+	}
+
+	// Set the endpoint if provided for testing
+	if d.config.AWSURL != "" {
+		d.dynamoDBClient = dynamodb.NewFromConfig(cfg, func(o *dynamodb.Options) {
+			o.EndpointResolverV2 = staticResolver{
+				BaseURL: d.config.AWSURL,
+			}
+		})
+	} else {
+		d.dynamoDBClient = dynamodb.NewFromConfig(cfg)
+	}
+
+	return nil
+}
+
+func (d *Destination) Write(ctx context.Context, records []opencdc.Record) (int, error) {
```

Comment:
Can we add an integration test for writing and deleting a few records?

Reply:
added